### PR TITLE
clib.session: Refactor to simplify the checking of currently open session

### PR DESCRIPTION
**Pull request by seisman**
Just some refactoring to simplify the codes. See the changes for more details.

### CODE REVIEW

File: pygmt/clib/session.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def session_pointer(self) -> ctp.c_void_p:
             If trying to access without a currently open GMT session (i.e., outside of
             the context manager).
         """
-        if not hasattr(self, "_session_pointer") or self._session_pointer is None:
+        if getattr(self, "_session_pointer", None) is None:
```
Comment (seisman) [inline pygmt/clib/session.py:163]: `if not hasattr(self, "_session_pointer") or self._session_pointer is None:` is equivalent to `if getattr(self, "_session_pointer", None) is None:`.

File: pygmt/clib/session.py [hunk 2]
```diff
@@ -338,19 +338,12 @@ def create(self, name: str):
         name
             A name for this session. Doesn't really affect the outcome.
         """
```
Comment (seisman) [inline pygmt/clib/session.py:340]: If `self.session_pointer` is defined and is not None, then we know we have an open session. The `try..except` statement was added in 2018 (#210) and is not necessary.

**Review by michaelgrund — APPROVED** [review]

File: pygmt/clib/session.py (hunk 2 above)
Comment (seisman) [reply pygmt/clib/session.py:340]: It's impossible that the `msg` assignment and the `raise GMTCLibError` lines can raise a `GMTCLibNoSessionError` exception, so we should be safe.

Comment (seisman) [reply pygmt/clib/session.py:340]: Reverted back to `try..except` in da9c5e17abdffbba49fc49bb9d82f49b2a794bc2 and improve the comments to make it easier to understand.

File: pygmt/clib/session.py [hunk 3]
```diff
@@ -338,17 +338,16 @@ def create(self, name: str):
...
+        # free to create a new one. Otherwise, we will raise a GMTCLibError exception.
         try:
-            # Won't raise an exception if there is a currently open session.
             _ = self.session_pointer
-            # In this case, fail to create a new session until the old one is destroyed.
             msg = (
-                "Failed to create a GMT API session: There is a currently open session."
-                " Must destroy it first."
+                "Failed to create a GMT API session: "
+                "There is a currently open session. Must destroy it first."
```
Comment (weiji14) [inline pygmt/clib/session.py:348]:
```suggestion
                "There is currently an open session. Must destroy it first."
```

**Review by weiji14 — APPROVED** [review]

**Pull request merged**